spec-main/api-app-path-spec.ts

```diff
@@ -0,0 +1,315 @@
+
+      expect(() => {
+        app.setPath('music', badPath);
+      }).to.throw(/Path must be absolute/);
+    });
+
+    it('does not create a new directory by default', () => {
+      const badPath = path.join(__dirname, 'music');
+
+      expect(fs.existsSync(badPath)).to.be.false();
+      app.setPath('music', badPath);
+      expect(fs.existsSync(badPath)).to.be.false();
+
+      expect(() => { app.getPath(badPath as any); }).to.throw();
+    });
+  });
+
+  describe('setAppLogsPath(path)', () => {
+    const appLogsPath = path.join(realtmpdir, 'mylogs');
+    it('by default', async () => {
+      const output = await runTestApp('app-custom-path');
+      switch (process.platform) {
+        case 'darwin':
+          expect(output.appLogs).to.equal(path.join(os.homedir(), 'Library', 'Logs', 'Electron'));
+          break;
+        case 'win32':
+        default:
+          expect(output.appLogs).to.equal(path.join(output.appData, defaultAppName, 'logs'));
+          expect(output.appLogs).to.equal(path.join(output.userData, 'logs'));
+          break;
+      }
+    });
+
+    // Linux
+    // app path module setAppLogsPath() setAppLogsPath(/tmp/mylogs) - setAppLogsPath(/tmp/mylogs)
+    // /home/builduser/project/src/electron/spec-main/api-app-path-spec.ts
+    // AssertionError: expected '/home/builduser/.config/app-custom-path/logs' to equal '/tmp/mylogs'
+
+    // if (process.platform === 'win32') {
+    //   it(`setAppLogsPath('${appLogsPath}')`, async () => {
+    //     const output = await runTestApp('app-custom-path', `-custom-applogs=${appLogsPath}`)
+    //     expect(output.appLogs).to.equal(appLogsPath)
+    //   })
+    // }
+
+    // To review later
+    it('setAppLogsPath()', async () => {
+      const defaultAppLogs = app.getPath('logs');
+      app.setAppLogsPath(appLogsPath);
+      expect(app.getPath('logs')).to.equal(appLogsPath);
+      app.setAppLogsPath();
+      expect(app.getPath('logs')).to.equal(defaultAppLogs);
+    });
+
+    it('throws when a relative path is passed', () => {
+      const badPath = 'hey/hi/hello';
+
+      expect(() => {
+        app.setAppLogsPath(badPath);
+      }).to.throw(/Path must be absolute/);
+    });
+  });
+
+  //  describe('getPath("logs")', () => {
+  //    const logsPaths = {
+  //      'darwin': path.resolve(os.homedir(), 'Library', 'Logs'),
+  //      'linux': path.resolve(app.getPath('userData'), 'logs'),
+  //      'win32': path.resolve(app.getPath('userData'), 'logs'),
+  //    }
+
+  //    it('has no logs directory by default', () => {
+  //      const osLogPath = (logsPaths as any)[process.platform]
+  //      expect(fs.existsSync(osLogPath)).to.be.false()
+  //    })
+
+  //    it('creates a new logs directory if one does not exist', () => {
+  //      expect(() => { app.getPath('logs') }).to.not.throw()
+
+  //      const osLogPath = (logsPaths as any)[process.platform]
+  //      expect(fs.existsSync(osLogPath)).to.be.true()
+  //    })
+  //  })
```

Comment:
These appear to be new tests but are commented out. Am I reading that right, and if so, what is your intent for these?

Reply:
intermediate tests, not needed, I removed them